src/Core/File/UploadedFile.php

```diff
@@ -0,0 +1,188 @@
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ *  @author 	PrestaShop SA <contact@prestashop.com>
+ *  @copyright  2007-2016 PrestaShop SA
+ *  @license    http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ *  International Registered Trademark & Property of PrestaShop SA
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Core\File;
+
+use ImageManager;
+use InvalidArgumentException;
+use PrestaShop\PrestaShop\Core\File\Exception\FileUploadException;
+use PrestaShop\PrestaShop\Core\File\Exception\InvalidFileException;
+use PrestaShop\PrestaShop\Core\File\Exception\MaximumSizeExceededException;
+
+class UploadedFile implements UploadedFileInterface
+{
+    /**
+     * @var int
+     */
+    protected $maximumSize;
+
+    /**
+     * @var string
+     */
+    protected $downloadDirectory;
+
+    /**
+     * @param string $downloadDirectory
+     * @param int $maximumSize
+     */
+    public function __construct(
+        string $downloadDirectory,
+        int $maximumSize
+    ) {
+        $this->downloadDirectory = $downloadDirectory;
+        $this->maximumSize = $maximumSize;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function upload($file): array
+    {
+        if (is_array($file)) {
+            return $this->uploadFromHttpPost($file);
+        }
+
+        if (is_string($file)) {
+            return $this->uploadFromBinaryFile($file);
+        }
+
+        throw new InvalidArgumentException();
+    }
+
+    /**
+     * Validate file size
+     *
+     * @param array $file
+     *
+     * @throws InvalidFileException
+     * @throws MaximumSizeExceededException
+     */
+    protected function validateSize(array $file): void
+    {
+        if (!isset($file['size'])) {
+            throw new InvalidFileException();
+        }
+
+        if ($file['size'] > $this->maximumSize) {
+            throw new MaximumSizeExceededException((string) $file['size']);
+        }
+    }
+
+    /**
+     * Validate if file is an uploaded file
+     *
+     * @param array $file
+     *
+     * @throws InvalidFileException
+     * @throws FileUploadException
+     */
+    protected function validateIsUploadedFile(array $file): void
+    {
+        if (!isset($file['tmp_name'])
+            || !isset($file['type'])
+            || !isset($file['name'])
+        ) {
+            throw new InvalidFileException();
+        }
+
+        if (!is_uploaded_file($file['tmp_name'])) {
+            throw new FileUploadException();
+        }
+    }
+
+    /**
+     * Generate file name from uniqid
+     *
+     * @return string
+     */
+    protected function generateFileName(): string
+    {
+        do {
+            $uniqid = sha1(uniqid()); // must be a sha1
+        } while (file_exists($this->downloadDirectory . $uniqid));
```

Comment:
Looks like a job for https://www.php.net/manual/en/function.tempnam.php

Reply:
Unfortunately, no because we want a sha1 as name, tempnam, we want to control the length of the name.